--- Slim/Formats/XML.pm.old	2009-02-05 08:06:17.000000000 +0000
+++ Slim/Formats/XML.pm	2009-02-05 08:06:17.000000000 +0000
@@ -1,6 +1,6 @@
 package Slim::Formats::XML;
 
-# $Id: XML.pm 24083 2008-11-25 17:56:36Z andy $
+# $Id$
 
 # Copyright 2006-2007 Logitech
 
@@ -13,7 +13,7 @@
 use strict;
 use File::Slurp;
 use HTML::Entities;
-use JSON::XS qw(from_json);
+use JSON::XS qw(decode_json);
 use Scalar::Util qw(weaken);
 use URI::Escape qw(uri_escape);
 use XML::Simple;
@@ -288,7 +288,7 @@
 	my $xml;
 	
 	if ( $type =~ /json/ ) {
-		$xml = from_json($$content);
+		$xml = decode_json($$content);
 	}
 	else {
 		$xml = xmlToHash($content);
