--- Slim/Utils/Accessor.pm.old	2009-06-27 18:37:44.000000000 +0100
+++ Slim/Utils/Accessor.pm	2009-06-27 18:38:08.000000000 +0100
@@ -94,7 +94,7 @@
 		if ($type eq 'rw') {
 			
 			if ( hasXS() ) {
-				Class::XSAccessor::Array::_generate_accessor(
+				Class::XSAccessor::Array::_generate_method(
 					$class,	$field,	$n, 0, 0, 'accessor',
 				);
 			}
@@ -108,7 +108,7 @@
 		} elsif ($type eq 'ro') {
 			
 			if ( hasXS() ) {
-				Class::XSAccessor::Array::_generate_accessor(
+				Class::XSAccessor::Array::_generate_method(
 					$class,	$field,	$n, 0, 0, 'getter',
 				);
 			}
